modules/apps/frontend-taglib/frontend-taglib-clay/src/main/resources/META-INF/resources/Checkbox.js

```diff
@@ -0,0 +1,37 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import {ClayCheckbox} from '@clayui/form';
+import React, {useState} from 'react';
+
+export default function Checkbox({
+	checked,
+	componentId: _componentId,
+	cssClass,
+	locale: _locale,
+	portletId: _portletId,
+	portletNamespace: _portletNamespace,
+	...otherProps
+}) {
+	const [value, setValue] = useState(checked);
+
+	return (
+		<ClayCheckbox
+			checked={value}
```

Comment:
According to the Clay Checkbox [docs](https://clayui.com/docs/components/checkbox.html) it needs to be a controlled component. What different props did u have in mind, did the name `value` cause the confusion?

Reply:
So, if `value` is `apple`, then we have `<input type="checkbox" checked="apple" />`?

Reply:
You think that the `value`  that gets set server-side would have bearing to this `value`  that gets defined by React's hook? I tested both with and without JS yesterday, with and without a `value` being set, and didn't discover any problems. But I guess it's better to be safe and rename this variable then from `value` to `isChecked`or similar.

Reply:
Ugh, sorry... I didn't even check. I wouldn't have imagined that we'd do `value = useState(checked)`... it feels like it's done specifically to confuse us :D 

That means that you can't pass a value, because it would be true, then, yes.

`const [isChecked, setChecked] = useState(checked)` sounds like the safer option. We even have occurrences of this in our code base, you can grep it ;)